### ISSUE TITLE
lstm with processing EEG data

### ISSUE THREAD

**Issue by sdustwdd** [fchollet/keras#774]
Now I have dataset of 10 objects ,the data is about sleep,30 seconds of data should be classified ,but   objects  have variable timing ,such as first has 616 the number of 30s,second has 745,.....
we want to use LSTM to classiy the state of sleep .
how can I format my data to the input that network needs.
What the network's struct

**Comment by holderm** [comment]
I think padding with zeros at the end of the (shorter) sequences is the key. Have a look at #85 maybe this solves your issue.

**Comment by sdustwdd** [comment]
AS above ,Do you think the number of samples is very small.
I studied imdb_lstm.py and found this:
1.First,the X_train is only 2-D matrix,and sequences' length are variable,so we should  pad_sequences(X_train, maxlen=maxlen) ,I think pad_sequences only adapt to 2-D matrix,Now I have 3-D matrix,so I padding with zeros at the end of the (shorter) sequences manually.
2. Second , transform 2-D matrix to 3-D matrix, so we use embedding layer,(20000,100)--->(20000,100,128),but my matrix(10,736,15000) so I needn't this step.
3. Third, LSTM,because I have 3-D matrix ,so I can use this layer firstly?I can use multi-LSTM , my struct of network as follow:

model = Sequential()

model.add(LSTM(15000, 128, activation='sigmoid', inner_activation='hard_sigmoid',return_sequences=True))
model.add(LSTM(128, 7, activation='sigmoid', inner_activation='hard_sigmoid',return_sequences=True))

model.add(Activation('softmax'))

model.compile(loss='binary_crossentropy', optimizer='rmsprop')

model.fit(new_set_X, to_categorical_y, batch_size=batch_size, nb_epoch=10,shuffle=False)
score = model.evaluate(new_set_X, to_categorical_y, batch_size=16,show_accuracy=True)

4.Fourth, one lstm layer maybe have nulti-block,and one block have multi-cell,but LSTM in this keras packages I have no idea about this ,
Do you think my understanding correct?

**Comment by ZMaple** [comment]
sdustwdd, have you solved your problem? I encountered the same problem and had no idea about how to mask the padding 3D matrix to LSTM. It seems the embedding layer does not accept 3D matrix as input. Am I right?

**Comment by lch507** [comment]
I also have similar work that I want to test LSTM as well. Please let us know if the padding solved the different data size issues. I also have a data that has different epoch sizes. I am trying to set the label as 2 or 3 (by adding some dummy label).
